bubblewrap.c

```diff
@@ -792,6 +812,93 @@ privileged_op (int         privileged_op_socket,
     }
 }
 
+struct _StringBuilder
+{
+  char * str;
+  size_t size;
+  size_t offset;
+};
+
+static void
+strappend(struct _StringBuilder *dest, const char *src)
+{
+  size_t len = strlen(src);
+  if (dest->offset + len >= dest->size) {
+    dest->size = (dest->size + len) * 2;
+    dest->str = realloc(dest->str, dest->size);
+    if (dest->str == NULL)
+      die ("Out of memory");
+  }
+
+  strcpy(dest->str + dest->offset, src);
```

Comment:
I am not comfortable with `strcpy` being used in a codebase that is sometimes setuid root. However careful you've been to allocate enough memory, and to avoid entering this code while privileged, it's just too easy to get wrong. As a minimum, this should use `strncpy` - but if it can be made to use things like `asprintf` then that would likely be safer.